### DISCUSSION TITLE
Did my "lovely" pet just ate the head off my engineer!?

### DISCUSSION THREAD

**Discussion by Benzin73** [FakeFishGames/Barotrauma#4886] [Ideas and suggestions]
During my current single player run, I decided to get a pet. After puting a saline on an egg, a "cute" little flying cthulu thing came out. Halloween pumpkin in hand and started flying arround the Sub. I fed it and a little heart poped on top of him. 

After a little while, Im cruising arround on my Orca, on my way to deliver some meds or something as part of a "Deliver XXX" mission, sonar is clear, no monsters near by, everything is going smoothly, untill suddenly, **My Engineer drops dead**. His health bar on the top left just turns red and dissapears. No messages on the chat box, no cries for help, Nothing. He just dies!
I was allready on the nav station steering the sub. I double check sonar, no contacts. No breeches on hull, no enemies onboard, no fires, **NOTHING**
So i run to the reactor room (where i asume my engineer would be), and as i enter the room, my Engineer is lying on the ground **and his head is missing!** There is no flooding, no monsters, **nothing**. Except... That flying little bastard cthulu.
He is not hostile to me or anything. He is just flying arorund next to the decapitated corpse of my engineer.

So, my question is, Did that flying little shit just ate the head of my Engineer???

I had a pet before, a little slime that crawled on the floor, and i had to drop it off at a station because he suddenly got a "taste" for chewing the testicles off 2 of my gunners. He wouldnt attack anyone else, except those 2 guards. He kept jumping at their waste and biting them, wich caused light injuries on them, but nothing serious. Certainly he dint decaptitate anyone!
Ibe read about a bug that causes pets to go hostile, but i would have never tought that my pet would chomp off the head of one of my crew.
Is this in fact a bug, or is it an intentional "surprise"?

**Comment by RadCatish** [comment]
I feel like you should open an issue report for this, it definitely doesn't sound like intended behaviour: https://github.com/Regalis11/Barotrauma/issues/new/choose

**Reply by Benzin73** [reply]
This is in fact the second time a pet has attacked my crew. I dint reported it because i tought it was a known bug, but i am gonna open a ticket, just in case the devs dont know about it.

**Comment by NilanthAnimosus** [comment]
I would be amazed and confused if this was the reason the engineer died. Specifically because the ragdoll XML of the cthulu pet is only capable of 0.1 strength stun and 1.0 strength (this is insignificant health loss by the way) bitewounds afflictions once per second as its only attack if it even got hostile, it also only having a structure damage of 0.

If the engineer was attacked by the cthulu pet, they should have declared intruders or such, or called for security to help - like they do when attacked by monsters or your own crew / yourself specifically if it was enough damage.

Instead what I think may have occurred is the engineer died to some other means (There is no guesswork for me as to how or why) and then perhaps, as I've not used pets yet but I have heard this may be a thing - the pet ate the dead engineers head (After they passed -100 negative health) as if the corpse was food (there is no distinction I imagine between a dead crawler and a dead human character, their both characters ultimately if this is a feature).

It is a shame you can no longer examine the body for what wounds they incurred to result in the character being dead, the cause of death likely would have helped the most.

**Reply by Benzin73** [reply]
Belive me, i was stunned myself! 
The first pet i had also attacked the crew (though they did not declared intruders, tried to fight back or anything. They just stood there taking the bites). But they certainly dint die from the pet attacks. Once they had been hurt enough, the medic would come and patch them up.
So imagine my surprise walking in and seeing my headless engineer. You are right, i should have examined the corpse to see what happened to it. But i just quit back to main menu.
If the pet did not attack the enigneer, then i have absolutely no clue of what happened. There was nothing in the room. No damage to it, the reactor was running fine, and there were no enemies outside (or inside).
Is there any chance the engineer tried to fix the reactor and got a deadly electrical shock? ibe never seen a character at full health "just die" from a single electrical shock.
